melodic/distribution.yaml

```diff
@@ -6,6 +6,14 @@ release_platforms:
   ubuntu:
   - bionic
 repositories:
+  ROS_driver-release:
+    release:
+      packages:
+      - pf_driver
+      tags:
+        release: release/melodic/{package}/{version}
+      url: https://github.com/PepperlFuchs/ROS_driver-release.git
```

Comment:
This repository does not appear to have a successful bloom release. I don't see the comments in the readme at the root, nor do there appear to be enough tags and branches in the release repository. https://github.com/PepperlFuchs/ROS_driver-release

Did you tell bloom to continue despite failures to resolve dependencies?

Reply:
I did not notice a failure during the bloom release. There was only a vcstools Warning, see below.

In the upstream repo we had 'master' as default branch, in the new -release repo the default branch is 'main'. Maybe this is the problem. We have now changed the default branch in the upstream repo to 'main'.
If you think that this solves the problem, I will start the bloom-release with a new repo name.

```batch
$ bloom-release --rosdistro melodic --track melodic ROS_driver-release --edit

ROS Distro index file associate with commit 'ee6e803c59a084864fd8b1227ece6a94e3693aff'
New ROS Distro index url: 'https://raw.githubusercontent.com/ros/rosdistro/ee6e803c59a084864fd8b1227ece6a94e3693aff/index-v4.yaml'
Specified repository 'ROS_driver-release' is not in the distribution file located at 'https://raw.githubusercontent.com/ros/rosdistro/ee6e803c59a084864fd8b1227ece6a94e3693aff/melodic/distribution.yaml'
Could not determine release repository url for repository 'ROS_driver-release' of distro 'melodic'
You can continue the release process by manually specifying the location of the RELEASE repository.
To be clear this is the url of the RELEASE repository not the upstream repository.
For release repositories on GitHub, you should provide the `https://` url which should end in `.git`.
Here is the url for a typical release repository on GitHub: https://github.com/ros-gbp/rviz-release.git
==> Looking for a release of this repository in a different distribution...
No reasonable default release repository url could be determined from previous releases.
Release repository url [press enter to abort]: https://github.com/PepperlFuchs/ROS_driver-release
The release repository url you provided does not end in `.git`.
Would you like to enter the address again [Y/n]? Y
Release repository url [press enter to abort]: https://github.com/PepperlFuchs/ROS_driver-release.git
==> Fetching 'ROS_driver-release' repository from 'https://github.com/PepperlFuchs/ROS_driver-release.git'
Cloning into '/tmp/tmpsHF7jZ'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (3/3), done.
WARNING [vcstools] Command failed: 'git checkout master'
 run at: '/tmp/tmpsHF7jZ'
 errcode: 1:
error: pathspec 'master' did not match any file(s) known to git.
[/vcstools]
Creating 'master' branch.
Creating track 'melodic'...
Repository Name:
  upstream
    Default value, leave this as upstream if you are unsure
  <name>
    Name of the repository (used in the archive name)
  ['upstream']: 
Upstream Repository URI:
  <uri>
    Any valid URI. This variable can be templated, for example an svn url
    can be templated as such: "https://svn.foo.com/foo/tags/foo-:{version}"
    where the :{version} token will be replaced with the version for this release.
  [None]: https://github.com/PepperlFuchs/ROS_driver.git
Upstream VCS Type:
  svn
    Upstream URI is a svn repository
  git
    Upstream URI is a git repository
  hg
    Upstream URI is a hg repository
  tar
    Upstream URI is a tarball
  ['git']: 
Version:
  :{ask}
    This means that the user will be prompted for the version each release.
    This also means that the upstream devel will be ignored.
  :{auto}
    This means the version will be guessed from the devel branch.
    This means that the devel branch must be set, the devel branch must exist,
    and there must be a valid package.xml in the upstream devel branch.
  <version>
    This will be the version used.
    It must be updated for each new upstream version.
  [':{auto}']: 
Release Tag:
  :{none}
    For svn and tar only you can set the release tag to :{none}, so that
    it is ignored.  For svn this means no revision number is used.
  :{ask}
    This means the user will be prompted for the release tag on each release.
  :{version}
    This means that the release tag will match the :{version} tag.
    This can be further templated, for example: "foo-:{version}" or "v:{version}"
    
    This can describe any vcs reference. For git that means {tag, branch, hash},
    for hg that means {tag, branch, hash}, for svn that means a revision number.
    For tar this value doubles as the sub directory (if the repository is
    in foo/ of the tar ball, putting foo here will cause the contents of
    foo/ to be imported to upstream instead of foo itself).
  [':{version}']: 
Upstream Devel Branch:
  <vcs reference>
    Branch in upstream repository on which to search for the version.
    This is used only when version is set to ':{auto}'.
  [None]: 
ROS Distro:
  <ROS distro>
    This can be any valid ROS distro, e.g. indigo, kinetic, lunar, melodic
  ['melodic']: 
Patches Directory:
  :{none}
    Use this if you want to disable overlaying of files.
  <path in bloom branch>
    This can be any valid relative path in the bloom branch. The contents
    of this folder will be overlaid onto the upstream branch after each
    import-upstream.  Additionally, any package.xml files found in the
    overlay will have the :{version} string replaced with the current
    version being released.
  [None]: 
Release Repository Push URL:
  :{none}
    This indicates that the default release url should be used.
  <url>
    (optional) Used when pushing to remote release repositories. This is only
    needed when the release uri which is in the rosdistro file is not writable.
    This is useful, for example, when a releaser would like to use a ssh url
    to push rather than a https:// url.
  [None]: 
Created 'melodic' track.
==> Testing for push permission on release repository
==> git remote -v
origin	git@github.com:PepperlFuchs/ROS_driver-release.git (fetch)
origin	git@github.com:PepperlFuchs/ROS_driver-release.git (push)
==> git push --dry-run
Everything up-to-date
==> Releasing 'ROS_driver-release' using release track 'melodic'
==> git-bloom-release melodic
Processing release track settings for 'melodic'
Checking upstream devel branch '<default>' for package.xml(s)
Cloning into '/tmp/tmpFqz1LN/upstream'...
remote: Enumerating objects: 182, done.
remote: Counting objects: 100% (182/182), done.
remote: Compressing objects: 100% (117/117), done.
remote: Total 1149 (delta 89), reused 126 (delta 54), pack-reused 967
Receiving objects: 100% (1149/1149), 260.93 KiB | 1.04 MiB/s, done.
Resolving deltas: 100% (651/651), done.
Looking for packages in 'master' branch... found 'pf_driver'.
Detected version '1.1.0' from package(s): ['pf_driver']

Executing release track 'melodic'
==> bloom-export-upstream /tmp/tmpFqz1LN/upstream git --tag 1.1.0 --display-uri https://github.com/PepperlFuchs/ROS_driver.git --name upstream --output-dir /tmp/tmpzZ2SBU
Checking out repository at 'https://github.com/PepperlFuchs/ROS_driver.git' to reference '1.1.0'.
Exporting to archive: '/tmp/tmpzZ2SBU/upstream-1.1.0.tar.gz'
md5: 406f3a85892da55945a5d724015b1b09
This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.

==> git-bloom-import-upstream /tmp/tmpzZ2SBU/upstream-1.1.0.tar.gz  --release-version 1.1.0 --replace
Creating upstream branch.
Importing archive into upstream branch...
Creating tag: 'upstream/1.1.0'
I'm happy.  You should be too.
This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.

==> git-bloom-generate -y rosrelease melodic --source upstream -i 1
Releasing package: ['pf_driver']
Releasing package 'pf_driver' for 'melodic' to: 'release/melodic/pf_driver'
This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.

==> git-bloom-generate -y rosdebian --prefix release/melodic melodic -i 1 --os-name ubuntu
Generating source debs for the packages: ['pf_driver']
Debian Incremental Version: 1
Debian Distributions: ['bionic']
Releasing for rosdistro: melodic

Pre-verifying Debian dependency keys...
Running 'rosdep update'...
All keys are OK

Placing debian template files into 'debian/melodic/pf_driver' branch.
==> Placing templates files in the 'debian' folder.

####
#### Generating 'bionic' debian for package 'pf_driver' at version '1.1.0-1'
####
Generating debian for bionic...
No historical releaser history, using current maintainer name and email for each versioned changelog entry.
No CHANGELOG.rst found for package 'pf_driver'
Package 'pf-driver' has dependencies:
Run Dependencies:
  rosdep key           => bionic key
  libpcl-all           => ['libpcl-apps1.8', 'libpcl-common1.8', 'libpcl-features1.8', 'libpcl-filters1.8', 'libpcl-io1.8', 'libpcl-kdtree1.8', 'libpcl-keypoints1.8', 'libpcl-ml1.8', 'libpcl-octree1.8', 'libpcl-outofcore1.8', 'libpcl-people1.8', 'libpcl-recognition1.8', 'libpcl-registration1.8', 'libpcl-sample-consensus1.8', 'libpcl-search1.8', 'libpcl-segmentation1.8', 'libpcl-stereo1.8', 'libpcl-surface1.8', 'libpcl-tracking1.8', 'libpcl-visualization1.8']
  message_runtime      => ['ros-melodic-message-runtime']
  rviz                 => ['ros-melodic-rviz']
  curlpp-dev           => ['libcurlpp-dev']
  dynamic_reconfigure  => ['ros-melodic-dynamic-reconfigure']
  laser_geometry       => ['ros-melodic-laser-geometry']
  libjsoncpp-dev       => ['libjsoncpp-dev']
  pcl_conversions      => ['ros-melodic-pcl-conversions']
  pcl_ros              => ['ros-melodic-pcl-ros']
  roscpp               => ['ros-melodic-roscpp']
  roscpp_serialization => ['ros-melodic-roscpp-serialization']
  sensor_msgs          => ['ros-melodic-sensor-msgs']
  std_msgs             => ['ros-melodic-std-msgs']
  tf                   => ['ros-melodic-tf']
Build and Build Tool Dependencies:
  rosdep key           => bionic key
  libpcl-all-dev       => ['libpcl-dev']
  message_generation   => ['ros-melodic-message-generation']
  roslint              => ['ros-melodic-roslint']
  curlpp-dev           => ['libcurlpp-dev']
  dynamic_reconfigure  => ['ros-melodic-dynamic-reconfigure']
  laser_geometry       => ['ros-melodic-laser-geometry']
  libjsoncpp-dev       => ['libjsoncpp-dev']
  pcl_conversions      => ['ros-melodic-pcl-conversions']
  pcl_ros              => ['ros-melodic-pcl-ros']
  roscpp               => ['ros-melodic-roscpp']
  roscpp_serialization => ['ros-melodic-roscpp-serialization']
  sensor_msgs          => ['ros-melodic-sensor-msgs']
  std_msgs             => ['ros-melodic-std-msgs']
  tf                   => ['ros-melodic-tf']
  catkin               => ['ros-melodic-catkin']
  rosunit              => ['ros-melodic-rosunit']
ROS Distro index file associate with commit 'ee6e803c59a084864fd8b1227ece6a94e3693aff'
New ROS Distro index url: 'https://raw.githubusercontent.com/ros/rosdistro/ee6e803c59a084864fd8b1227ece6a94e3693aff/index-v4.yaml'
==> In place processing templates in 'debian' folder.
Expanding 'debian/changelog.em' -> 'debian/changelog'
Expanding 'debian/control.em' -> 'debian/control'
Expanding 'debian/source/options.em' -> 'debian/source/options'
Expanding 'debian/source/format.em' -> 'debian/source/format'
Expanding 'debian/copyright.em' -> 'debian/copyright'
Expanding 'debian/gbp.conf.em' -> 'debian/gbp.conf'
Expanding 'debian/rules.em' -> 'debian/rules'
Expanding 'debian/compat.em' -> 'debian/compat'
Creating tag: debian/ros-melodic-pf-driver_1.1.0-1_bionic
####
#### Successfully generated 'bionic' debian for package 'pf_driver' at version '1.1.0-1'
####

This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.

==> git-bloom-generate -y rosdebian --prefix release/melodic melodic -i 1 --os-name debian --os-not-required
No platforms defined for os 'debian' in release file for the 'melodic' distro. This os was not required; continuing without error.
This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.

==> git-bloom-generate -y rosrpm --prefix release/melodic melodic -i 1 --os-name fedora
No platforms defined for os 'fedora' in release file for the 'melodic' distro.
Not performing RPM generation.
This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.

==> git-bloom-generate -y rosrpm --prefix release/melodic melodic -i 1 --os-name rhel
No platforms defined for os 'rhel' in release file for the 'melodic' distro.
Not performing RPM generation.
This version of bloom is '0.10.0', but the newest available version is '0.10.1'. Please update.




Tip: Check to ensure that the debian tags created have the same version as the upstream version you are releasing.
Everything went as expected, you should check that the new tags match your expectations, and then push to the release repo with:
  git push --all && git push --tags  # You might have to add --force to the second command if you are over-writing existing tags
<== Released 'ROS_driver-release' using release track 'melodic' successfully
==> git remote -v
origin	git@github.com:PepperlFuchs/ROS_driver-release.git (fetch)
origin	git@github.com:PepperlFuchs/ROS_driver-release.git (push)
Releasing complete, push to release repository?
Continue [Y/n]? Y
==> Pushing changes to release repository for 'ROS_driver-release'
==> git push --all
Counting objects: 138, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (117/117), done.
Writing objects: 100% (138/138), 60.64 KiB | 2.42 MiB/s, done.
Total 138 (delta 17), reused 0 (delta 0)
remote: Resolving deltas: 100% (17/17), done.
To github.com:PepperlFuchs/ROS_driver-release.git
 * [new branch]      debian/melodic/bionic/pf_driver -> debian/melodic/bionic/pf_driver
 * [new branch]      debian/melodic/pf_driver -> debian/melodic/pf_driver
 * [new branch]      master -> master
 * [new branch]      patches/debian/melodic/bionic/pf_driver -> patches/debian/melodic/bionic/pf_driver
 * [new branch]      patches/debian/melodic/pf_driver -> patches/debian/melodic/pf_driver
 * [new branch]      patches/release/melodic/pf_driver -> patches/release/melodic/pf_driver
 * [new branch]      release/melodic/pf_driver -> release/melodic/pf_driver
 * [new branch]      upstream -> upstream
<== Pushed changes successfully
==> Pushing tags to release repository for 'ROS_driver-release'
==> git push --tags
Total 0 (delta 0), reused 0 (delta 0)
To github.com:PepperlFuchs/ROS_driver-release.git
 * [new tag]         debian/ros-melodic-pf-driver_1.1.0-1_bionic -> debian/ros-melodic-pf-driver_1.1.0-1_bionic
 * [new tag]         release/melodic/pf_driver/1.1.0-1 -> release/melodic/pf_driver/1.1.0-1
 * [new tag]         upstream/1.1.0 -> upstream/1.1.0
<== Pushed tags successfully
==> Generating pull request to distro file located at 'https://raw.githubusercontent.com/ros/rosdistro/ee6e803c59a084864fd8b1227ece6a94e3693aff/melodic/distribution.yaml'
Would you like to add documentation information for this repository? [Y/n]? n
Would you like to add source information for this repository? [Y/n]? n
Would you like to add a maintenance status for this repository? [Y/n]? n
Unified diff for the ROS distro file located at '/tmp/tmpFG_v_U/ROS_driver-release-1.1.0-1.patch':
--- ee6e803c59a084864fd8b1227ece6a94e3693aff/melodic/distribution.yaml
+++ ee6e803c59a084864fd8b1227ece6a94e3693aff/melodic/distribution.yaml
@@ -6,6 +6,14 @@
   ubuntu:
   - bionic
 repositories:
+  ROS_driver-release:
+    release:
+      packages:
+      - pf_driver
+      tags:
+        release: release/melodic/{package}/{version}
+      url: https://github.com/PepperlFuchs/ROS_driver-release.git
+      version: 1.1.0-1
   abb:
     release:
       packages:
==> Checking on GitHub for a fork to make the pull request from...
Could not find a fork of ros/rosdistro on the PepperlFuchs GitHub account.
Would you like to create one now?
Continue [Y/n]? y
==> Using this fork to make a pull request from: PepperlFuchs/rosdistro
==> Cloning PepperlFuchs/rosdistro...
==> mkdir -p rosdistro
==> git init
Initialized empty Git repository in /tmp/1xmluL/rosdistro/.git/
Pull Request Title: ROS_driver-release: 1.1.0-1 in 'melodic/distribution.yaml' [bloom]
Pull Request Body : 
Increasing version of package(s) in repository `ROS_driver-release` to `1.1.0-1`:

- upstream repository: https://github.com/PepperlFuchs/ROS_driver.git
- release repository: https://github.com/PepperlFuchs/ROS_driver-release.git
- distro file: `melodic/distribution.yaml`
- bloom version: `0.10.0`
- previous version for package: `null`

Open a pull request from 'PepperlFuchs/rosdistro:bloom-ROS_driver-release-0' into 'ros/rosdistro:master'?
Continue [Y/n]? y
==> git checkout -b bloom-ROS_driver-release-0
Switched to a new branch 'bloom-ROS_driver-release-0'
==> Pulling latest rosdistro branch
remote: Enumerating objects: 154345, done.
remote: Total 154345 (delta 0), reused 0 (delta 0), pack-reused 154345
Receiving objects: 100% (154345/154345), 92.49 MiB | 16.97 MiB/s, done.
Resolving deltas: 100% (93749/93749), done.
From https://github.com/ros/rosdistro
 * branch                master     -> FETCH_HEAD
==> git reset --hard ee6e803c59a084864fd8b1227ece6a94e3693aff
HEAD is now at ee6e803c5 bno055: 0.1.1-3 in 'foxy/distribution.yaml' [bloom] (#28267)
==> Writing new distribution file: melodic/distribution.yaml
==> git add melodic/distribution.yaml
==> git commit -m "ROS_driver-release: 1.1.0-1 in 'melodic/distribution.yaml' [bloom]"
[bloom-ROS_driver-release-0 766c173d3] ROS_driver-release: 1.1.0-1 in 'melodic/distribution.yaml' [bloom]
 1 file changed, 8 insertions(+)
==> Pushing changes to fork
Counting objects: 4, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (4/4), 479 bytes | 479.00 KiB/s, done.
Total 4 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
remote: 
remote: Create a pull request for 'bloom-ROS_driver-release-0' on GitHub by visiting:
remote:      https://github.com/PepperlFuchs/rosdistro/pull/new/bloom-ROS_driver-release-0
remote: 
To https://github.com/PepperlFuchs/rosdistro.git
 * [new branch]          bloom-ROS_driver-release-0 -> bloom-ROS_driver-release-0
<== Pull request opened at: https://github.com/ros/rosdistro/pull/28271
```